src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/CosmosTemplate.java

```diff
@@ -422,19 +374,18 @@ public void deleteById(String collectionName, Object id, PartitionKey partitionK
         feedOptions.populateQueryMetrics(isPopulateQueryMetrics);
 
         final SqlQuerySpec sqlQuerySpec = new FindQuerySpecGenerator().generateCosmos(query);
-        final FeedResponse<CosmosItemProperties> feedResponse =
-                cosmosClient.getDatabase(this.databaseName)
-                .getContainer(collectionName)
-                .queryItems(sqlQuerySpec, feedOptions)
-                .doOnNext(propertiesFeedResponse -> fillAndProcessResponseDiagnostics(responseDiagnosticsProcessor,
-                    null, propertiesFeedResponse))
-                .next()
-                .block();
-
-        if (feedResponse == null) {
-            throw new CosmosDBAccessException("Failed to query documents");
-        }
+        final FeedResponse<CosmosItemProperties> feedResponse = cosmosClient
+            .getDatabase(this.databaseName)
+            .getContainer(collectionName)
+            .queryItems(sqlQuerySpec, feedOptions)
+            .doOnNext(propertiesFeedResponse -> fillAndProcessResponseDiagnostics(responseDiagnosticsProcessor,
+                null, propertiesFeedResponse))
+            .onErrorResume(throwable ->
+                exceptionHandler("Failed to query items", throwable))
+            .next()
+            .block();
 
+        assert feedResponse != null;
```

Comment:
Not sure if feedResponse cannot be null here?

Reply:
The feedResponse cannot be null here.